src/executor/graph_executor.cc

```diff
@@ -355,6 +355,21 @@ Graph AssignContext(Graph g,
       vcontext.push_back(ctx_list[assigned_device[i]]);
     }
   }
+
+  // after device planning, we should check again
+  // if the assigned device of gradient node
+  // corresponds to storage of grads
+  auto &new_idx = g.indexed_graph();
+  for (size_t i = num_forward_outputs; i < g.outputs.size(); ++i) {
+    const uint32_t nid = new_idx.outputs()[i].node_id;
+    Context ctx = arg_grad_ctxes[i - num_forward_outputs];
+    CHECK(ctx == vcontext[nid])
```

Comment:
This is doing same checking as in line 340, just checking all gradient nodes, but at that time we haven't assign device to the gradient node, so this inconsistency is not detected, which will cause later crash.